components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/kafka/KAFKAMessageListener.java

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright (c) 2005-2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.inbound.endpoint.protocol.kafka;
+
+import kafka.consumer.*;
+import org.I0Itec.zkclient.exception.ZkTimeoutException;
+import org.apache.synapse.SynapseException;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class KAFKAMessageListener extends AbstractKafkaMessageListener {
+
+    public KAFKAMessageListener(int threadCount, List<String> topics,
+                                Properties kafkaProperties, InjectHandler injectHandler)
+            throws Exception {
+        this.threadCount = threadCount;
+        this.topics = topics;
+        this.kafkaProperties = kafkaProperties;
+        this.injectHandler = injectHandler;
+    }
+
+    /**
+     * Create the connection with the zookeeper to consume the messages
+     */
+    public boolean createKafkaConsumerConnector() throws Exception {
+
+        logger.debug("Create the connection and start to consume the streams");
+        boolean isCreated;
+        try {
+            if (consumerConnector == null) {
+                logger.info("Creating Kafka Consumer Connector...");
+                consumerConnector = Consumer
+                        .createJavaConsumerConnector(new ConsumerConfig(
+                                kafkaProperties));
+                logger.info("Kafka Consumer Connector is created");
+                start();
+
+            }
+            isCreated = true;
+        } catch (ZkTimeoutException toe) {
+            logger.error(" Error in Creating Kafka Consumer Connector | ZkTimeout"
```

Comment:
I follow the following standard in upcoming PRs.

catch (NumberFormatException nfe) {
                handleException("the count should be a number.", nfe);
  }

private void handleException(String msg, Exception ex) {
       log.error(msg, ex);
       throw new SynapseException(ex);
    }